### PR TITLE
Switch to stream-based loading

### PR DESCRIPTION
**Pull request by koppor**
Uses `ProtectedTermsLoader.class.getResourceAsStream` instead of `ProtectedTermsLoader.class.getResource`.

Refs https://github.com/JabRef/jabref/pull/11298#issuecomment-2121172277

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java [hunk 1]
```diff
@@ -29,30 +30,37 @@ public class ProtectedTermsParser {
...
-        readTermsList(path);
-    }
-
-    private void readTermsList(Path path) {
+        path = path.toAbsolutePath();
         if (!Files.exists(path)) {
             LOGGER.warn("Could not read terms from file {}", path);
             return;
         }
-        try (Stream<String> lines = Files.lines(path, StandardCharsets.UTF_8)) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java:52]: Files lines does not work with graal? Tbh this looks like a major drawback here.

Comment (calixtus) [reply src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java:52]: BufferedReader#lines seems to be somewhat of an equivalent here...

Comment (koppor) [reply src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java:52]: In line 60, I use exactly this function.

This call is only a wrapper for the call if a user supplies a file!

GraalVM does not support `.class.getResource(` with hacks. See https://github.com/oracle/graal/issues/7682 for a longer discussion. Especially https://github.com/oracle/graal/issues/7682#issuecomment-1786704111

> The reason NativeImageFileSystem behaves like JarFileSystem, not like UnixFileSystem, is that, unlike UnixFileSystem, you can't, for example, delete a file (resource) on both NativeImageFileSystem and JarFileSystem. The resources in NativeImageFileSystem are baked into the read-only section of the image heap

**Comment by koppor** [comment]
... and the tests fail... OMG

>  Test readProtectedTermsListFromFileReadsDescription() FAILED

File: src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java [hunk 2]
```diff
@@ -29,30 +30,37 @@ public class ProtectedTermsParser {
...
-        try {
-            Path path = Path.of(Objects.requireNonNull(ProtectedTermsLoader.class.getResource(Objects.requireNonNull(resourceFileName))).toURI());
-            readTermsList(path);
-            description = descriptionString;
-            location = resourceFileName;
-        } catch (URISyntaxException e1) {
-            LOGGER.error("");
+        description = descriptionString;
+        location = resourceFileName;
+        try (InputStream inputStream = ProtectedTermsLoader.class.getResourceAsStream(Objects.requireNonNull(resourceFileName))) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java:35]: You need to use getClassLoader() first

Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java:35]:
```suggestion
        try (InputStream inputStream = ProtectedTermsLoader.class.getClassLoader().getResourceAsStream(Objects.requireNonNull(resourceFileName))) {
```

Comment (koppor) [reply src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java:35]: Works without here.

**Comment by koppor** [comment]
For full GraalVM, we would maybe to patch `CSL.java`:

```java
            // try to find style in classpath
            url = CSL.class.getResource(styleName);
```

However, I added a workaround, because this branch is only taken if the provided String is not an XML file.

**Comment by koppor** [comment]
Assuming that JavaFX can perfectly deal with URLs coming from the classpath, we have one class left to "really" change:

org.jabref.logic.citationstyle.CitationStyle#discoverCitationStyles: Here, the list could be generated during compilation time

File: src/main/java/org/jabref/logic/importer/util/MathMLParser.java [hunk 3]
```diff
@@ -38,7 +41,7 @@ public static String parse(XMLStreamReader reader) {
             Source xmlSource = new StreamSource(new StringReader(xmlContent));
 
             // No SystemId required, because no relative URLs need to be resolved
-            Source xsltSource = new StreamSource(xsltResource);
+            Source xsltSource = new StreamSource(xsltResource, MathMLParser.class.getResource(XSLT_FILE_PATH).toExternalForm());
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/util/MathMLParser.java:44]: OK for a workaround. - Can we do add a fake path here? Maybe, #future work

**Comment by koppor** [comment]
Proposal: Get this in to see whether there are issues. As follow up the "difficult" conversions.

File: src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java (hunk 2 above)
Comment (koppor) [reply src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java:35]: I tried with `./gradlew run` and then protected terms. Worked.

**Comment by Siedlerchr** [comment]
Use the binary to test it!

**Comment by koppor** [comment]
Tested. Protected terms works. Think, more users should test it and we can revert if needed ^^.

**Review by calixtus — APPROVED** [review]
Lets move forward here.

**Pull request merged**